community/shell/src/main/java/org/neo4j/shell/impl/AbstractAppServer.java

```diff
@@ -52,21 +52,30 @@
 {
     private final Map<String, App> apps = new TreeMap<>();
 
+    public AbstractAppServer()
+            throws RemoteException
+    {
+        this( true );
+    }
+
 	/**
 	 * Constructs a new server.
 	 * @throws RemoteException if there's an RMI error.
 	 */
-	public AbstractAppServer()
-		throws RemoteException
-	{
-		super();
-        for ( App app : Service.load( App.class ) )
+    public AbstractAppServer( boolean addFromServiceLoading )
+        throws RemoteException
+    {
+        super();
+        if ( addFromServiceLoading )
         {
-            addApp( app );
+            for ( App app : Service.load( App.class ) )
+            {
+                addApp( app );
+            }
         }
-	}
+    }
 
-    private void addApp( App app )
+    protected void addApp( App app )
```

Comment:
For other shell app servers to add their own `App` instances. Like if I'm writing a custom shell tool I'm going to need this.